test/vmtests/vmtests/utils/libvirt_vm.py

```diff
@@ -94,6 +103,47 @@ def close(self) -> None:
         except libvirt.libvirtError as ex:
             logging.warning(f"VM delete failed. {ex}")
 
+
+    def create_ssh_client(
+        self,
+        ssh_private_key_path: Path,
+        test_temp_dir: Path,
+    ) -> SshClient:
+
+        ssh_known_hosts_path = test_temp_dir.joinpath("known_hosts")
+        open(ssh_known_hosts_path, "w").close()
+
+        # arm64 emulated runs take a very long time to boot and get to a state
+        # where we can connect to it.
+        ip_wait_time = 30
+        if platform.machine() == 'aarch64':
+            ip_wait_time = 300
+
+        # For arm64 runs, we are seeing a behavior where the first IP address that
+        # gets assigned becomes unusable by the time we try to ssh into the machine
+        # and then ssh fails to connect.
+        # Some time later, a different IP address gets assigned, and that IP
+        # address is usable.
+        stable_ip_time_out = 360
+        stable_ip_wait_time = 120
+        stable_ip_start_time = datetime.now()
```

Comment:
Should probably use `time.monotonic()` here.

Reply:
pending

Reply:
fixed.